packages/superset-ui-core/README.md

```diff
@@ -0,0 +1,90 @@
+## `@superset-ui/core`
+
+[![Version](https://img.shields.io/npm/v/@superset-ui/core.svg?style=flat)](https://img.shields.io/npm/v/@superset-ui/core.svg?style=flat)
+
+Core modules for Superset:
+
+- `SupersetClient` requests and authentication
+- (future) `i18n` locales and translation
+
+### SupersetClient
+
+The `SupersetClient` handles all client-side requests to the Superset backend. It can be configured
+for use within the Superset application, or used to issue `CORS` requests in other applications. At
+a high-level it supports:
+
+- `CSRF` token authentication
+  - queues requests in the case that another request is made before the token is received
+  - it checks for a token before every request, an external app that uses this can detect this by
+    catching errors, or explicitly checking `SupersetClient.isAuthorized()`
+- supports `GET` and `POST` requests (no `PUT` or `DELETE`)
+- timeouts
+- query aborts through the `AbortController` API
+
+#### Example usage
+
+```javascript
+// appSetup.js
+import { SupersetClient } from `@superset-ui/core`;
+// or import SupersetClient from `@superset-ui/core/lib|esm/SupersetClient`;
+
+SupersetClient.configure(...clientConfig);
+SupersetClient.init(); // CSRF auth, can also chain `.configure().init();
+
+// anotherFile.js
+import { SupersetClient } from `@superset-ui/core`;
+
+SupersetClient.post(...requestConfig)
+  .then(({ request, json }) => ...)
+  .catch((error) => ...);
+```
+
+#### API
+
+##### Client Configuration
+
+The following flags can be passed in the client config call
+`SupersetClient.configure(...clientConfig);`
+
+- `protocol = 'http'`
+- `host`
+- `headers`
+- `credentials = 'same-origin'` (set to `include` for non-Superset apps)
+- `mode = 'same-origin'` (set to `cors` for non-Superset apps)
+- `timeout`
+
+##### Per-request Configuration
+
+The following flags can be passed on a per-request call `SupersetClient.get/post(...requestConfig);`
+
+- `url` or `endpoint`
+- `headers`
+- `body`
+- `timeout`
+- `signal` (for aborting, from `const { signal } = (new AbortController())`)
+- for `POST` requests
+  - `postPayload` (key values are added to a `new FormData()`)
+  - `stringify` whether to call `JSON.stringify` on `postPayload` values
+
+##### Query aborting
+
+Query aborting is implemented through the `AbortController` API:
+
+```javascript
+import { SupersetClient } from '@superset-ui/core';
+import AbortController from 'abortcontroller-polyfill';
+
+const controller = new AbortController();
+const { signal } = controller;
+
+SupersetClient.get({}).then(...).catch(...);
+
+if (IWantToCancelForSomeReason) {
+  signal.abort(); // Promise is rejected, request `catch` is invoked
```

Comment:
Does it cancel every outgoing request? Is there a way to cancel just one?

Reply:
oops I forgot an important part of this usage (I'll update), you pass the signal to the request, so it's per-request aborting. 

see [here](https://github.com/apache/incubator-superset/pull/5772/files#diff-66f9c9696a5b417d0f13a79c9f1d3a7fR141) for an example in superset.